portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,56 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Mycale!</h1>");
+    private List<String> hardList;
+
+    @Override
+    public void init() {
+
+        hardList = new ArrayList<>(); 
+
+        /* hardList.add("Huston, we don't have a problem");
+        hardList.add("Hello! My name is DataServlet, and I will be your server for tonight");
+        hardList.add("Reach for the skyyyyyy"); */
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+
+        Gson gson = new Gson();
+        String json = gson.toJson(hardList);
+
+
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String text = getParameter(request, "comments-input", "");
+        hardList.add(text + " ");
+
+        Gson gson = new Gson();
+
+        response.setContentType("application/json");
+        String json = gson.toJson(hardList);
+
+        response.getWriter().println(json);
+        response.sendRedirect("/index.html");
+    }
+
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Rather than making this return a String with an explicitly passed in default value, consider returning an Optional<String> instead.

This lets you do thinks like call .orElse("") directly at the function callsite, making it very explicit the behavior you expect.